### PR TITLE
PauseOverlay/FailOverlay refactor

### PR DESCRIPTION
**Pull request by EVAST9919**

### CODE REVIEW

**Comment by peppy** [comment]
why are you so insistent on moving everything you possible can to `Player.cs`?

File: osu.Game/Screens/Play/Player.cs [hunk 1]
```diff
@@ -33,14 +32,16 @@ public class Player : OsuScreen
...
-        internal override bool HasLocalCursorDisplayed => !hasReplayLoaded && !IsPaused;
+        internal override bool HasLocalCursorDisplayed => !hasReplayLoaded && !IsPaused && !IsFailed;
 
         private bool hasReplayLoaded => hitRenderer.InputManager.ReplayInputHandler != null;
 
         public BeatmapInfo BeatmapInfo;
 
         public bool IsPaused { get; private set; }
 
+        public bool IsFailed { get; private set; }
```
Comment (Jorolf) [inline osu.Game/Screens/Play/Player.cs]: I don't know if this is a kind of code convention but shouldn't it be `HasFailed`

File: osu.Game/Screens/Play/InGameOverlay.cs [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+                    Width = 1f
+                }
+            };
+
+            Retries = 0;
+
+            AddButtons(colours);
+        }
+
+        protected virtual void AddButtons(OsuColour colours)
```
Comment (Jorolf) [inline osu.Game/Screens/Play/InGameOverlay.cs]: why don't you make this abstract?

File: osu.Desktop.VisualTests/Tests/TestCaseInGameOverlays.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+            failOverlay = new FailOverlay
+            {
+                Depth = -1,
+                OnRetry = () => Logger.Log(@"Retry"),
+                OnQuit = () => Logger.Log(@"Quit"),
+            };
+
+            Add(pauseOverlay);
+            Add(failOverlay);
```
Comment (Jorolf) [inline osu.Desktop.VisualTests/Tests/TestCaseInGameOverlays.cs]: You _could_ do this inline 
`Add(pauseoverlay = new PauseOverlay { ... });`

**Pull request merged**